src/instance_launcher.ts

```diff
@@ -77,8 +77,43 @@ export default class InstanceLauncher {
             if (count < group.scalingOptions.desiredCount && count < group.scalingOptions.maxDesired) {
                 ctx.logger.info('[Launcher] Will scale up to the desired count', { groupName, desiredCount, count });
 
-                const actualScaleUpQuantity =
+                let actualScaleUpQuantity =
                     Math.min(group.scalingOptions.maxDesired, group.scalingOptions.desiredCount) - count;
+
+                // if untracked throttle enabled, only scale up if there aren't too many untracked instances
+                if (group.enableUntrackedThrottle == null || group.enableUntrackedThrottle == true) {
+                    // use desired scaleUpQuantity to ensure we only scale up this many (plus one) until the previous batch are ready
+                    const untrackedThrottleThreshold = group.scalingOptions.scaleUpQuantity + 1;
+                    const untrackedCount = await this.audit.getGroupUntrackedCount(ctx, group.name);
+                    const allowedScaleUp = untrackedThrottleThreshold - untrackedCount;
```

Comment:
I am having some second thoughts, as it seems that even when we have no untracked items, which is the majority of the situations, we will never be able to launch at one time more than the scaleUpQuantity on the group, which I think it's not something we intend. We wanted an autoscaler that reacts fast.
Another option could be: if untracked count > untrackedThreshold, fire error and do not launch. Otherwise, allow the launch; in case of problems, soon that threshold will be reached and the launcher will stop (and we will alarm based on it).

Reply:
Upon review, we decided on a global max value which will be used if the group size max is greater than the global max.